src/manager-6.spec.jest.js

```diff
@@ -0,0 +1,409 @@
+      delete window.headertag;
+      adManager.fetchAmazonBids(slots);
+      expect(googlePushSpy).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  // Setup method utilized by #refreshSlot & #refreshSlots tests
+  function adSlotSetup() {
+    var baseContainer, container1, adSlot1, stubSlot, eagerStubSlot, variableReferences;
+    baseContainer = document.createElement('div');
+    container1 = document.createElement('div');
+    container1.className ='expected';
+    container1.id = 'ad-container-1';
+    adSlot1 = document.createElement('div');
+    adSlot1.id = 'dfp-ad-1';
+    adSlot1.className = 'dfp';
+    adSlot1.dataset.adUnit = 'header';
+    container1.appendChild(adSlot1);
+    baseContainer.appendChild(container1);
+    document.body.appendChild(baseContainer);
+
+    stubSlot = {
+      element: adSlot1,
+      prebid: 1,
+      activeSizes: [[300, 250]],
+      getSlotElementId: function() {
+        return adSlot1.id;
+      },
+      getTargeting: function () {
+        return '';
+      },
+      setTargeting: function () {},
+      getOutOfPage: function () { return false; }
+    };
+
+    eagerStubSlot = Object.assign({}, stubSlot, { eagerLoad: true });
+
+    variableReferences = {
+      baseContainer: baseContainer,
+      container1: container1,
+      adSlot1: adSlot1,
+      stubSlot: stubSlot,
+      eagerStubSlot: eagerStubSlot
+    };
+
+    adManager.slots = {
+      'dfp-ad-1': stubSlot
+    };
+
+    return variableReferences;
+  }
+
+  describe('#fetchIasTargeting', function(){
+    afterEach(function(){
+      adManager.__iasPET.queue = [];
+    });
+
+    it('- pushes ad slots to PET tag queue', function(){
+      adManager = AdManagerWrapper.init({
+        iasEnabled: true
+      });
+
+      adSlotSetup();
+
+      adManager.fetchIasTargeting();
+
+      expect(Array.isArray(adManager.__iasPET.queue)).toEqual(true);
+      expect(adManager.__iasPET.queue.length).toEqual(1)
+    });
+  });
+
+  describe('#refreshSlot', function() {
+    var adSlot, stubSlot, baseContainer;
+
+    beforeEach(function() {
+      var setupRefs = adSlotSetup();
+      adSlot = setupRefs.adSlot1;
+      stubSlot = setupRefs.stubSlot;
+      baseContainer = setupRefs.baseContainer;
+      adManager.fetchAmazonBids = jest.fn();
+      window.apstag = jest.fn();
+    });
+
+    afterEach(function() {
+      $(baseContainer).remove();
+    });
+
+    it('- loads the DFP slot matching up with the DOM element id', function() {
+      var refreshSpy = jest.spyOn(adManager, 'refreshSlots');
+      adManager.refreshSlot(adSlot);
+      expect(refreshSpy).toHaveBeenCalledWith([stubSlot]);
+    });
+
+    it('- should invoke optional callback onRefresh if provided', function () {
+      var spy = jest.spyOn(adManager.adUnits.units.header, 'onRefresh');
+      adManager.refreshSlot(adSlot);
+      expect(spy).toHaveBeenCalled();
+    });
+
+    it('- calls fetchAmazonBids if parameter amazonEnabled is true', function() {
```

Comment:
I don't think this test was even running before, looks like a bad merge: https://github.com/theonion/bulbs-public-ads-manager/blob/15762303cc7c56c61ab9bcc23224d6717820613c/src/manager.spec.js#L1172 but moved here and using `refreshSlot` instead of `initGoogleTag` to test because the latter eventually calls the former